suite-common/wallet-core/src/fiat-rates/fiatRatesMiddleware.ts

```diff
@@ -80,6 +82,26 @@ export const prepareFiatRatesMiddleware = createMiddlewareWithExtraDeps(
             }
         }
 
+        // Fetch fiat rates for all tokens of newly suite-native discovered account.
+        if (accountsActions.createIndexLabeledAccount.match(action)) {
+            const localCurrency = selectLocalCurrency(getState());
+
+            const { tokens, symbol } = action.payload;
+            tokens?.forEach(token => {
+                dispatch(
+                    updateFiatRatesThunk({
+                        ticker: {
+                            symbol,
+                            tokenAddress: token.contract as TokenAddress,
+                        },
+                        rateType: 'current',
+                        localCurrency,
+                        lastSuccessfulFetchTimestamp: Date.now() as Timestamp,
```

Comment:
Isn't this lie? There is no previous fetch at all :D

Reply:
Okey was looking to it and we probably should never have a property `lastSuccessfulFetchTimestamp` saying `Date.now()` in `updateFiatRatesThunk` but instead put it inside reducer the `Date.now()`.

Reply:
@AdamSchinzel removed the `lastSuccessfulFetchTimestamp` from thunk in the commit [a3551a5](https://github.com/trezor/trezor-suite/pull/11682/commits/a3551a5cd48db7163482b0fd41b94f18361104cb).

@Nodonisko Could you review again, please?

Reply:
`Date.now()` inside a redux reducer is an anti-pattern. Reducers should be pure functions and shouldn't call functions like this from inside of them. You should pass this inside an action creator to let the caller handle this functionality.

Reply:
Maybe it's fine but naming is bad, it should something like `fetchAttemptTimestamp` or something like that because at moment we are calleding this we have no idea if it will be successful.

Reply:
And @juriczech is right, reducer should be pure. Just rename that param of thunk to something that is true :D And in reducer name should stay same because there it's actually mean `lastSuccessfulFetchTimestamp`

Reply:
fe3542754bf3bb5c258deb4e7a7fcf2130a252fc